read-fonts/src/offset_array.rs

```diff
@@ -0,0 +1,136 @@
+//! Arrays of offsets with dynamic resolution
+//!
+//! This module provides a number of types that wrap arrays of offsets, dynamically
+//! resolving individual offsets as they are accessed.
+
+use crate::offset::ResolveNullableOffset;
+use font_types::{BigEndian, Nullable, Offset16, Scalar};
+
+use crate::{FontData, FontReadWithArgs, Offset, ReadArgs, ReadError, ResolveOffset};
+
+/// An array of offsets that can be resolved on access.
```

Comment:
I suggest explaining here what this types job is, why we can't just give you the slice of offsets directly. To my understanding the answer is we could, but then the caller would have to pass in data and read args to resolve items which would be ugly.